### PR TITLE
Steady state pipe elevation constraint

### PR DESCRIPTION
**Pull request by hskkanth**
Added a constraint for inclined pipes.
Updated the relevant files to include "elevation" as a column in junctions.
Created a function "ref_theta" to add inclination angles for pipes
Updated the files to include "g" = 9.81 as a global constant.
Updated the weymouth constraint template to take care of zero diameter pipes.

### CODE REVIEW

**Comment by kaarthiksundar** [comment]
@hskkanth Do you need to get g as an input parameter. @rb004f Let me know what your thoughts are on this. It is never going to change right. Otherwise it seems okay.

File: src/form/wp.jl [hunk 1]
```diff
@@ -30,6 +30,15 @@ function constraint_pipe_weymouth(gm::AbstractWPModel, n::Int, k, i, j, f_min, f
     end
 end
 
+"Pressure drop equation for inclined pipes"
```
Comment (pseudocubic) [inline src/form/wp.jl:33]: Can you add the equation to the docstring?

File: src/core/data.jl [hunk 2]
```diff
@@ -1298,6 +1298,24 @@ function _calc_pipe_resistance(pipe::Dict{String,Any}, base_length, base_pressur
     return resistance
 end
 
+"Calculates the constants required for the pressure drop along inclined pipes.
+Refer to the following paper for the derivation of the equation:
+S.K.K. Hari et al., Operation of Natural Gas Pipeline Networks With Storage Under Transient Flow Conditions"
```
Comment (pseudocubic) [inline src/core/data.jl:1303]: Can you add the equations to the docstring

File: src/io/matgas.jl [hunk 3]
```diff
@@ -43,6 +44,7 @@ const _mg_sources_columns = Vector{Tuple{String,Type}}([
 
 const _mg_junction_columns = Vector{Tuple{String,Type}}([
     ("id", Int),
+    ("elevation", Float64),
```
[the remark this answers is not in the document]
Comment (hskkanth) [reply src/io/matgas.jl]: Updated the rest of the changes. Missing elevation data is taken care of in ref_pipe_theta.

File: src/io/gaslib.jl [hunk 4]
```diff
@@ -348,7 +348,7 @@ function _get_junction_entry(junction)
     lon_sym = :geoWGS84Long in keys(junction) ? :geoWGS84Long : :y
     lon = parse(Float64, junction[lon_sym])
 
-    height = parse(Float64, junction["height"][:value])
+    elevation = parse(Float64, junction["height"][:value])
```
Comment (pseudocubic) [inline src/io/gaslib.jl:351]: should not `junction["height"][:value]` be `junction["elevation"][:value]` instead?

**Review by pseudocubic — CHANGES_REQUESTED** [review]

File: src/io/gaslib.jl (hunk 4 above)
Comment (hskkanth) [reply src/io/gaslib.jl:351]: Height is a gaslib property. So I converted to elevation only after reading it.

**Review by pseudocubic — APPROVED** [review]

**Pull request merged**